src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -354,6 +371,14 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                         // SetData because repeating appconfig calls (by not updating watchers) won't help anything for keyvault calls.
                         // As long as adapter.NeedsRefresh is true, we will attempt to update keyvault again the next time RefreshAsync is called.
                         SetData(await PrepareData(applicationSettings, cancellationToken).ConfigureAwait(false));
+                        if (logDebugBuilder.Length > 0)

```

Comment:
`SetData` may throw exceptions unrelated to the key-values and feature flags being watched. So we should log whatever we have so far before calling SetData.